bundles/org.openhab.binding.hapero/src/main/java/org/openhab/binding/hapero/internal/device/Device.java

```diff
@@ -0,0 +1,151 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.hapero.internal.device;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link Device} class contains the raw
+ * data received from the heating control system and
+ * conversion functions.
+ *
+ * @author Daniel Walter - Initial contribution
+ */
+@NonNullByDefault
+public class Device {
+    /** Raw data stream from the device */
+    protected String[] dataItems = {};
+    private final Logger logger = LoggerFactory.getLogger(Device.class);
+
+    /**
+     * Constructor
+     *
+     * @param items List of data items as Strings
+     */
+    public Device(String[] items) {
+        dataItems = items;
+    }
+
+    /**
+     * Return the data item at the given index
+     *
+     * @return A String containing the data item at the Index or "" if the Index is invalid
+     */
+    public String getDataItem(int index) {
+        if (index >= 0 && index < dataItems.length) {
+            return dataItems[index];
+        } else {
+            logger.warn("getDataItem: Invalid Index {}.", String.valueOf(index));
```

Comment:
I think you do not need to cast index, the logger will do it for you.

Reply:
OK, changed it.